apollo-federation/src/query_graph/extract_subgraphs_from_supergraph.rs

```diff
@@ -1317,15 +1422,25 @@ fn add_subgraph_field(
     };
 
     for argument in &field.arguments {
+        let mut destination_argument = InputValueDefinition {
+            description: None,
+            name: argument.name.clone(),
+            ty: argument.ty.clone(),
+            default_value: argument.default_value.clone(),
+            directives: Default::default(),
+        };
+        if let Some(cost_spec_definition) = cost_spec_definition {
+            cost_spec_definition.propagate_demand_control_directives(
+                &subgraph.schema,
+                &argument.directives,
+                &mut destination_argument.directives,
+                original_directive_names,
+            )?;
+        }
+
         subgraph_field
             .arguments
-            .push(Node::new(InputValueDefinition {
-                description: None,
-                name: argument.name.clone(),
-                ty: argument.ty.clone(),
-                default_value: argument.default_value.clone(),
-                directives: Default::default(),
-            }))
+            .push(Node::new(destination_argument))
```

Comment:
I found that I was doing a lot of novel work for directive extraction in the federation repo because the extraction logic really hasn't been implemented for a lot of directives, even if they are allowed on arguments. So here we handle each of the arguments, and the very similar logic below is to copy over the directives on the field itself.